src/Symfony/Component/Config/Builder/ConfigBuilderGenerator.php

```diff
@@ -172,17 +172,18 @@ private function handlePrototypedArrayNode(PrototypedArrayNode $node, ClassBuild
         if (null !== $parameterType || $prototype instanceof ScalarNode) {
             $property = $class->addProperty($node->getName());
             if (null === $key = $node->getKeyAttribute()) {
+                // This is a array of scalar values. Dont use "singular name".
                 $body = '
 /**
```

Comment:
You can in fact write (I had to look it up):

```php
$rootNode
    ->children()
        ->arrayNode('floats')
            ->prototype('float')->end()
            ->example([3.1415])
        ->end()
    ->end()
;
```

But that is super super rare. 99.9% of the cases uses `scalar`, `array` and `variable`. Can we write any valuable doc block for these types? Will the added complexity be worth it? 

| Prototype type | Doc block |
| -- | -- |
| scalar | `@param list $value`
| array | `@param array $value`
| variable | `@param list\|array $value`
| float | `@param list<float> $value`
| boolean | `@param list<bool> $value`
| integer | `@param list<int> $value`
| enum | `@param list<enum> $value`

Reply:
actually, the typo for a prototyped array node without a key attribute is always `list<T>`. The prototype type only controls the value of `T`. As we are able to provide a more precise type for the argument for such nodes, we are able to provide a more precise type for `T` (the empty string becoming `mixed` in phpdoc).

Reply:
> actually, the type for a prototyped array node without a key attribute is always list<T>

Are you sure about that? Example is the `metadata` for [workflows](
https://github.com/symfony/symfony/blob/3f0f21c3619f36fe22fba6d9e800258f3bc3252a/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php#L526-L531). It is using an array node in a prototyp array. That will not be a `list<T>`, it will be an `array`. 

Since this is just a comment, I'll leave that feature for a future update.

Reply:
a prototype node without key with an array prototype is a `list<array>`

Reply:
So, I do `list<mixed>` 99.9% of the time. If I know it is a `array`, `float`, etc I do `list<array>`, `list<float>`, etc.

Reply:
@Nyholm I think that 99% of `ScalarNode` usages are actually expecting a `StringNode` that does not exist

Reply:
Yes. I am sure you are correct. But technically, we dont know that. ¯\_(ツ)_/¯

Reply:
what I mean is that if we want to improve that PHP config layer in the future, maybe we should introduce an actual StringNode enforcing that the value is a string rather than a scalar